### PR TITLE
fix noJit builds

### LINKED ISSUE #112: Daily builds of DisableJIT (x86, x64) are broken.
See table at: https://github.com/Microsoft/ChakraCore/wiki/Build-Status#disablejit-builds

Confirmed the compile error locally.

```
E:\dev\ChakraCore\lib\Runtime\Language\JavascriptStackWalker.h(229): error C3646: 'GetCachedInternalFrameInfo': unknown override specifier [E:\dev\ChakraCore\lib\Runtime\Base\Chakra.Runtime.Base.vcxproj]
E:\dev\ChakraCore\lib\Runtime\Language\JavascriptStackWalker.h(229): error C2059: syntax error: '(' [E:\dev\ChakraCore\lib\Runtime\Base\Chakra.Runtime.Base.vcxproj]
E:\dev\ChakraCore\lib\Runtime\Language\JavascriptStackWalker.h(229): error C2334: unexpected token(s) preceding '{'; skipping apparent function body [E:\dev\ChakraCore\lib\Runtime\Base\Chakra.Runtime.Base.vcxproj]
E:\dev\ChakraCore\lib\Runtime\Language\JavascriptStackWalker.h(330): error C3646: 'lastInternalFrameInfo': unknown override specifier [E:\dev\ChakraCore\lib\Runtime\Base\Chakra.Runtime.Base.vcxproj]
E:\dev\ChakraCore\lib\Runtime\Language\JavascriptStackWalker.h(330): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [E:\dev\ChakraCore\lib\Runtime\Base\Chakra.Runtime.Base.vcxproj]
```

Looks like it might be related to https://github.com/Microsoft/ChakraCore/commit/c6c4de4104526e4cb5203a73394d3345a2fa0476 by @rajatd

### PR DESCRIPTION
**Pull request by rajatd**
fix noJit builds

Closes #112

### CODE REVIEW

**Comment by rajatd** [comment]
@dilijev can you review this?

**Comment by dilijev** [comment]
@dotnet-bot test nojit please

File: lib/Runtime/Language/JavascriptStackWalker.cpp [hunk 1]
```diff
@@ -818,6 +825,7 @@ namespace Js
                 Assert(this->lastInternalFrameInfo.codeAddress);
                 this->lastInternalFrameInfo.frameConsumed = true;
             }
+#endif
```
Comment (dilijev) [inline lib/Runtime/Language/JavascriptStackWalker.cpp]: To make it more clear can you change this to `#endif // ENABLE_NATIVE_CODEGEN` and the above `#endif` to `#endif // DEBUG`?

File: lib/Runtime/Language/JavascriptStackWalker.cpp [hunk 2]
```diff
@@ -1023,6 +1032,7 @@ namespace Js
         }
         this->lastInternalFrameInfo.loopBodyFrameType = loopBodyFrameType;
     }
+#endif
```
Comment (dilijev) [inline lib/Runtime/Language/JavascriptStackWalker.cpp]: Same  `#endif // ENABLE_NATIVE_CODEGEN` here

File: lib/Runtime/Language/JavascriptStackWalker.cpp (hunk 1 above)
Comment (rajatd) [inline lib/Runtime/Language/JavascriptStackWalker.cpp]: Sure. nested #ifdefs- can get confusing

File: lib/Runtime/Language/JavascriptStackWalker.cpp (hunk 2 above)
Comment (rajatd) [inline lib/Runtime/Language/JavascriptStackWalker.cpp]: This seems pretty clear as to which #if it corresponds to

**Comment by dilijev** [comment]
Jenkins is running slowly. I'm confirming locally that the change merged with master compiles and passes tests.

**Comment by dilijev** [comment]
Looks like the test script is looking for the NoJit binaries in the wrong place for the `-disableJit` config. That may cause the disablejit checks I triggered to fail. I'll commit an updated script. This change builds, so I think this change will be fine as-is, once it can merge with the new test script.

File: lib/Runtime/Language/JavascriptStackWalker.cpp (hunk 2 above)
Comment (dilijev) [inline lib/Runtime/Language/JavascriptStackWalker.cpp]: True. This one is fine.

**Comment by dilijev** [comment]
@dotnet-bot test nojit please

**Comment by dilijev** [comment]
LGTM barring checks and the test script issue I mentioned.

**Comment by davkean** [comment]
Please stop queuing more builds of the same changes - all you are doing adding more builds to the queue, making Jenkins take even longer.

Jenkins is a shared service between many teams, so be mindful of resources.

//cc @mmitche

**Comment by dilijev** [comment]
@davkean @mmitche I understand your frustration. We feel it too. In this case I don't believe we're abusing the resource. This change affects the dailies so we need them to run to verify its correctness. As for the extra requeue of all checks, I apologize. I should have reviewed the change first and waited for the update before adding the dailies.

**Comment by dilijev** [comment]
@rajatd All tests pass on my machine when this change is merged on top of #117.

**Comment by dilijev** [comment]
@davkean @mmitche By the way, I've been wondering if there's an easy way to cancel all queued requests that haven't been issued build machines for a given PR? When a new commit is added to a PR and checks re-queue, do the old ones that haven't run yet get cancelled?

**Comment by mmitche** [comment]
It doesn't automatically cancel, but looking at the source I'm led to believe it may be able to be done.

Sent from my Windows Phone
________________________________

**Comment by dilijev** [comment]
I cancelled the daily checks that were already triggered on this one as I believe they will fail until #117 gets merged in. I'll re-queue the dailies once that's done.

**Comment by davkean** [comment]
@dilijev I was referring to these:

image

I thought these were duplicated because of your mention double mention of the trigger phrase for your nojit tests, but I was wrong. Apologies. Looking at them, I can't figure out why both sets were triggered:

image

@mmitche Maybe this is just Jenkins/Github catching up with the original request, and then queue again up when the PR branch was updated? They are so close to each other in time.

**Comment by dilijev** [comment]
@rajatd You're good to merge this one one tests pass.

**Comment by dilijev** [comment]
@dotnet-bot test dailies please

**Comment by dilijev** [comment]
@digitalinfinity Do we currently support ARM builds for the disablejit configuration? Just want to know whether they were supposed to work.

**Comment by dilijev** [comment]
@rajatd I accidentally triggered all the dailies instead of just the relevant ones. `Windows 7 daily_x64_debug` is a known failing configuration at the moment. I'll trigger another one and then kill it so at least this PR will be yellow instead of red.

**Comment by dilijev** [comment]
@dotnet-bot test Windows 7 daily_x64_debug please

**Comment by rajatd** [comment]
@dilijev , do we have any historical data pertaining to state of arm builds with no-jit?

**Comment by dilijev** [comment]
@rajatd Not according to Jenkins. @digitalinfinity was working on this before so that's why I tagged him. I guess we could try the commit at the time Hitesh originally merged in the addition of the NoJIT configurations and see what happens there.

**Comment by dilijev** [comment]
I tried from that commit with the current jenkins configs I recently corrected.

`jenkins.buildone.cmd arm release "/p:BuildJIT=false"`

Produces errors:

```
Chakra.Runtime.Language.lib(arm_Thunks.obj) : error LNK2001: unresolved external symbol "private: static void * (__cdecl*__cdecl
Js::DynamicProfileInfo::EnsureDynamicProfileInfo(class Js::ScriptFunction *))(class Js::RecyclableObject *,struct Js::CallInfo,..
.)" (?EnsureDynamicProfileInfo@DynamicProfileInfo@Js@@CAP6APAXPAVRecyclableObject@2@UCallInfo@2@ZZPAVScriptFunction@2@@Z) [E:\dev
\Chakra3\core\bin\ChakraCore\ChakraCore.vcxproj]
Chakra.Runtime.Language.lib(arm_Thunks.obj) : error LNK2001: unresolved external symbol "private: static void * (__cdecl*__cdecl
Js::InterpreterStackFrame::EnsureDynamicInterpreterThunk(class Js::ScriptFunction *))(class Js::RecyclableObject *,struct Js::Cal
lInfo,...)" (?EnsureDynamicInterpreterThunk@InterpreterStackFrame@Js@@CAP6APAXPAVRecyclableObject@2@UCallInfo@2@ZZPAVScriptFuncti
on@2@@Z) [E:\dev\Chakra3\core\bin\ChakraCore\ChakraCore.vcxproj]
E:\dev\Chakra3\core\Build\VcBuild\bin\arm_release\ChakraCore.dll : fatal error LNK1120: 2 unresolved externals [E:\dev\Chakra3\co
re\bin\ChakraCore\ChakraCore.vcxproj]
```

**Comment by dilijev** [comment]
Same error as currently.  I think your change is good to check in as-is and we'll follow up with the ARM issue in a separate PR.

**Comment by dilijev** [comment]
Committed as https://github.com/Microsoft/ChakraCore/commit/a0f58b0f499e488ed69bde93570229e169f488d1

**Comment by rajatd** [comment]
Thanks @dilijev , I was waiting to see the build status turn green

**Comment by dilijev** [comment]
@rajatd It was green before I marked as merged. That's how I knew it was in.

**Pull request closed without merging**